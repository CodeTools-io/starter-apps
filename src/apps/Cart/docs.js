import React from 'react';
import { describe, PropTypes } from 'react-desc';
import { DocsPage } from 'internal/components/Docs';
import Cart from './Cart';

const CartWithSchema = describe(Cart).description(
  'A feature for showing items that a customer intends to purchase.'
);

CartWithSchema.displayName = 'Cart';

CartWithSchema.propTypes = {
  children: PropTypes.element.description(
    'The components to show inside of the main area of the Cart.'
  ),
};

export default () => {
  return <DocsPage category="apps" component={CartWithSchema} />;
};
