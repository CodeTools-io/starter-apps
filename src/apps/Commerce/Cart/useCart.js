import { useCallback, useMemo, useState } from 'react';
import accounting from 'accounting';
import { cart } from 'data';

export default function useCart() {
  const [products, setProducts] = useState(cart.products);

  const lineItems = useMemo(() => {
    return products.map((product) => {
      const total = product.quantity * product.price;
      return {
        ...product,
        priceLabel: accounting.formatMoney(product.price),
        total,
        totalLabel: accounting.formatMoney(total),
      };
    });
  }, [products]);

  const total = useMemo(() => {
    return lineItems.reduce((accum, lineItem) => accum + lineItem.total, 0);
  }, [lineItems]);

  const totalLabel = useMemo(() => {
    return accounting.formatMoney(total);
  }, [total]);

  const removeProduct = useCallback(
    (id) => {
      setProducts(products.filter((product) => product.id !== id));
    },
    [products]
  );
  const updateProduct = useCallback(
    (payload) => {
      setProducts(
        products.map((product) => {
          if (product.id !== payload.id) {
            return product;
          }

          return {
            ...product,
            ...payload,
          };
        })
      );
    },
    [products]
  );

  return useMemo(() => {
    return {
      lineItems,
      products,
      total,
      totalLabel,
      updateProduct,
      removeProduct,
    };
  }, [lineItems, products, total, totalLabel, updateProduct, removeProduct]);
}
