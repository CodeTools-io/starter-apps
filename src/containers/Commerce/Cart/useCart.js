import { useMemo, useState } from 'react';
import accounting from 'accounting';
const { PUBLIC_URL } = process.env;

export default function useCart() {
  const [products] = useState([
    {
      id: 'product-1',
      title: 'Coffee',
      description:
        'Reprehenderit in sint eiusmod esse Lorem laboris cillum nostrud ut quis ipsum non duis magna. Ea sit non id qui minim do nisi tempor ad. Proident ipsum aliqua nisi magna reprehenderit deserunt exercitation veniam laborum ad est aliquip amet. Culpa ullamco id esse culpa exercitation. Velit amet aliqua cillum tempor dolor laborum laborum. Tempor do officia ex enim anim. Labore do qui est occaecat fugiat et anim id laboris.',
      image: `${PUBLIC_URL}/placeholder/img/food/coffee-1200xauto.jpg`,
      price: 10.99,
      quantity: 3,
    },
    {
      id: 'product-2',
      title: 'Honey',
      description:
        'Reprehenderit in sint eiusmod esse Lorem laboris cillum nostrud ut quis ipsum non duis magna. Ea sit non id qui minim do nisi tempor ad. Proident ipsum aliqua nisi magna reprehenderit deserunt exercitation veniam laborum ad est aliquip amet. Culpa ullamco id esse culpa exercitation. Velit amet aliqua cillum tempor dolor laborum laborum. Tempor do officia ex enim anim. Labore do qui est occaecat fugiat et anim id laboris.',
      image: `${PUBLIC_URL}/placeholder/img/food/honey-1200xauto.jpg`,
      price: 200.99,
      quantity: 3,
    },
    {
      id: 'product-3',
      title: 'Onions',
      description:
        'Reprehenderit in sint eiusmod esse Lorem laboris cillum nostrud ut quis ipsum non duis magna. Ea sit non id qui minim do nisi tempor ad. Proident ipsum aliqua nisi magna reprehenderit deserunt exercitation veniam laborum ad est aliquip amet. Culpa ullamco id esse culpa exercitation. Velit amet aliqua cillum tempor dolor laborum laborum. Tempor do officia ex enim anim. Labore do qui est occaecat fugiat et anim id laboris.',
      image: `${PUBLIC_URL}/placeholder/img/food/onion-1200xauto.jpg`,
      price: 3000.99,
      quantity: 1,
    },
    {
      id: 'product-5',
      title: 'Strawberries',
      description:
        'Reprehenderit in sint eiusmod esse Lorem laboris cillum nostrud ut quis ipsum non duis magna. Ea sit non id qui minim do nisi tempor ad. Proident ipsum aliqua nisi magna reprehenderit deserunt exercitation veniam laborum ad est aliquip amet. Culpa ullamco id esse culpa exercitation. Velit amet aliqua cillum tempor dolor laborum laborum. Tempor do officia ex enim anim. Labore do qui est occaecat fugiat et anim id laboris.',
      image: `${PUBLIC_URL}/placeholder/img/food/strawberry-1200xauto.jpg`,
      price: 500.99,
      quantity: 7,
    },
    {
      id: 'product-6',
      title: 'Tea',
      description:
        'Reprehenderit in sint eiusmod esse Lorem laboris cillum nostrud ut quis ipsum non duis magna. Ea sit non id qui minim do nisi tempor ad. Proident ipsum aliqua nisi magna reprehenderit deserunt exercitation veniam laborum ad est aliquip amet. Culpa ullamco id esse culpa exercitation. Velit amet aliqua cillum tempor dolor laborum laborum. Tempor do officia ex enim anim. Labore do qui est occaecat fugiat et anim id laboris.',
      image: `${PUBLIC_URL}/placeholder/img/food/tea-1200xauto.jpg`,
      price: 6000.99,
      quantity: 2,
    },
  ]);

  const lineItems = useMemo(() => {
    return products.map((product) => {
      const total = product.quantity * product.price;
      return {
        ...product,
        priceLabel: accounting.formatMoney(product.price),
        total,
        totalLabel: accounting.formatMoney(total),
      };
    });
  }, [products]);

  const total = useMemo(() => {
    return lineItems.reduce((accum, lineItem) => accum + lineItem.total, 0);
  }, [lineItems]);

  const totalLabel = useMemo(() => {
    return accounting.formatMoney(total);
  }, [total]);

  const store = useMemo(() => {
    return {
      lineItems,
      products,
      total,
      totalLabel,
    };
  }, [lineItems, products, total, totalLabel]);

  return store;
}
