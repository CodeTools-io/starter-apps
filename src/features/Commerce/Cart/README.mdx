---
type: component
id: Cart
moduleId: Commerce
name: Cart
categoryId: feature
description: A feature for showing items that a customer is about to purchase.
path: /cart
domain:
  - name: Cart
    description: A collection of products that will be purchased when the user is done shopping and ready to pay.
  - name: Line Item
    description: A specific entry in a cart, which includes a product, product pricing, purchase quantity, and total cost for the number of products being purchased.
  - name: Product
    description: An item a user intends to purchase.
grommet:
  - Box
  - Button
  - Image
  - Paragraph
  - Table
  - TableHeader
  - TableCell
  - TableRow
  - TableBody
  - TableFooter
  - Text
  - TextInput
---

## Overview

A feature for showing items that a customer is about to purchase. This example demonstrates how Boxes, Forms, and Tables can cooperatively display and update state variables. It also shows how to abstract out data elements and how to build components that have other components inside.
