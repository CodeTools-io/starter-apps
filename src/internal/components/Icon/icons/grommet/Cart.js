
import React from 'react';
import { Cart } from 'grommet-icons';

export default function CartIcon({...props}) {
  return <Cart {...props}/>
}
  