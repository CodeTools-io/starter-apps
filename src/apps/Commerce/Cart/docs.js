import React from 'react';
import { describe, PropTypes } from 'react-desc';
import { DocsPage } from 'components/Docs';
import Cart from './Cart';
const usage = `
<Cart>
  // ...rest of the app.
<Cart>
`;
const CartWithSchema = describe(Cart)
  .description('An app for showing items that a customer intends to purchase.')
  .usage(usage);

CartWithSchema.propTypes = {
  children: PropTypes.element.description(
    'The components to show inside of the main area of the Cart.'
  ),
};

export default () => {
  return <DocsPage component={CartWithSchema} />;
};
