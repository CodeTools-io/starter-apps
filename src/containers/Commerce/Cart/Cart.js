import React from 'react';
import {
  Box,
  Button,
  Card,
  Image,
  Paragraph,
  Table,
  TableHeader,
  TableCell,
  TableRow,
  TableBody,
  TableFooter,
  Text,
  TextInput,
} from 'grommet';
import { FormClose } from 'grommet-icons';
import useCart from './useCart';

export default function Cart({ children, ...props }) {
  const { lineItems, totalLabel, removeProduct, updateProduct } = useCart();

  return (
    <Box className="Cart" pad="medium" fill>
      <Card background="white" pad="medium">
        <Table>
          <TableHeader>
            <TableRow>
              <TableCell scope="col"></TableCell>
              <TableCell scope="col"></TableCell>
              <TableCell scope="col">
                <Text weight="bold">Product</Text>
              </TableCell>
              <TableCell scope="col">
                <Text weight="bold">Price</Text>
              </TableCell>
              <TableCell scope="col">
                <Text weight="bold">Quantity</Text>
              </TableCell>
              <TableCell scope="col">
                <Text weight="bold">Total</Text>
              </TableCell>
            </TableRow>
          </TableHeader>
          <TableBody>
            {lineItems.map((lineItem) => (
              <TableRow key={`cart-line-item-${lineItem.id}`}>
                <TableCell scope="row">
                  <Button
                    icon={<FormClose />}
                    onClick={() => removeProduct(lineItem.id)}
                  />
                </TableCell>
                <TableCell scope="row">
                  <Box height="small" width="small">
                    <Image src={lineItem.image} fit="contain" />
                  </Box>
                </TableCell>
                <TableCell scope="row">
                  <Text weight="bold">{lineItem.title}</Text>
                  <Paragraph color="dark-6">{lineItem.description}</Paragraph>
                </TableCell>
                <TableCell scope="row">
                  <Text weight="bold">{lineItem.priceLabel}</Text>
                </TableCell>
                <TableCell scope="row">
                  <Box width="xsmall">
                    <TextInput
                      type="number"
                      value={lineItem.quantity}
                      onChange={(event) => {
                        updateProduct({
                          id: lineItem.id,
                          quantity: event.target.value,
                        });
                      }}
                    />
                  </Box>
                </TableCell>
                <TableCell scope="row">
                  <Text weight="bold">{lineItem.totalLabel}</Text>
                </TableCell>
              </TableRow>
            ))}
          </TableBody>
          <TableFooter>
            <TableRow>
              <TableCell></TableCell>
              <TableCell></TableCell>
              <TableCell></TableCell>
              <TableCell></TableCell>
              <TableCell align="right">
                <Text weight="bold">Total</Text>
              </TableCell>
              <TableCell>
                <Text weight="bold">{totalLabel}</Text>
              </TableCell>
            </TableRow>
          </TableFooter>
        </Table>
      </Card>
    </Box>
  );
}
