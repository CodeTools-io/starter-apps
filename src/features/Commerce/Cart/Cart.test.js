import React from 'react';
import Cart from './Cart';

describe('Cart', () => {
  test('can render', () => {
    const { container } = renderContainer(
      <Cart>
        <p>Content for the Cart</p>
      </Cart>
    );

    expect(container).toMatchSnapshot();
  });
});
