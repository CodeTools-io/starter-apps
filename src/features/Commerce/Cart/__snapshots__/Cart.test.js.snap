// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart can render 1`] = `
<div>
  <div
    class="ErrorBoundary"
  >
    <div
      class="StyledGrommet-sc-19lkkz7-0 ggTDHw"
    >
      <table
        class="StyledTable-sc-1m3u5g-6 icFZHB"
      >
        <thead
          class="StyledTable__StyledTableHeader-sc-1m3u5g-4"
        >
          <tr
            class="StyledTable__StyledTableRow-sc-1m3u5g-2"
          >
            <th
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 kuHWcz"
              scope="col"
            />
            <th
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 kuHWcz"
              scope="col"
            />
            <th
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 kuHWcz"
              scope="col"
            >
              <span
                class="StyledText-sc-1sadyjn-0 fiywiB"
              >
                Product
              </span>
            </th>
            <th
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 kuHWcz"
              scope="col"
            >
              <span
                class="StyledText-sc-1sadyjn-0 fiywiB"
              >
                Price
              </span>
            </th>
            <th
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 kuHWcz"
              scope="col"
            >
              <span
                class="StyledText-sc-1sadyjn-0 fiywiB"
              >
                Quantity
              </span>
            </th>
            <th
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 kuHWcz"
              scope="col"
            >
              <span
                class="StyledText-sc-1sadyjn-0 fiywiB"
              >
                Total
              </span>
            </th>
          </tr>
        </thead>
        <tbody
          class="StyledTable__StyledTableBody-sc-1m3u5g-3"
        >
          <tr
            class="StyledTable__StyledTableRow-sc-1m3u5g-2"
          >
            <th
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 jbJGGF"
              scope="row"
            >
              <button
                class="StyledButton-sc-323bzc-0 dYCPbF"
                type="button"
              >
                <svg
                  aria-label="FormClose"
                  class="StyledIcon-ofa7kd-0 gkyWxk"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7,7 L17,17 M7,17 L17,7"
                    fill="none"
                    stroke="#000"
                    stroke-width="2"
                  />
                </svg>
              </button>
            </th>
            <th
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 jbJGGF"
              scope="row"
            >
              <div
                class="StyledBox-sc-13pk1d4-0 ldZMaz"
              >
                <img
                  class="StyledImage-ey4zx9-0 dfToQT"
                  src="/placeholder/img/food/coffee-1200xauto.jpg"
                />
              </div>
            </th>
            <th
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 jbJGGF"
              scope="row"
            >
              <span
                class="StyledText-sc-1sadyjn-0 fiywiB"
              >
                Coffee
              </span>
              <p
                class="StyledParagraph-tbetod-0 ca-DKMR"
              >
                Reprehenderit in sint eiusmod esse Lorem laboris cillum nostrud ut quis ipsum non duis magna. Ea sit non id qui minim do nisi tempor ad. Proident ipsum aliqua nisi magna reprehenderit deserunt exercitation veniam laborum ad est aliquip amet. Culpa ullamco id esse culpa exercitation. Velit amet aliqua cillum tempor dolor laborum laborum. Tempor do officia ex enim anim. Labore do qui est occaecat fugiat et anim id laboris.
              </p>
            </th>
            <th
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 jbJGGF"
              scope="row"
            >
              <span
                class="StyledText-sc-1sadyjn-0 fiywiB"
              >
                $10.99
              </span>
            </th>
            <th
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 jbJGGF"
              scope="row"
            >
              <div
                class="StyledBox-sc-13pk1d4-0 degYya"
              >
                <div
                  class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 flNVco"
                >
                  <input
                    autocomplete="off"
                    class="StyledTextInput-sc-1x30a0s-0 kGmGrs"
                    type="number"
                    value="3"
                  />
                </div>
              </div>
            </th>
            <th
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 jbJGGF"
              scope="row"
            >
              <span
                class="StyledText-sc-1sadyjn-0 fiywiB"
              >
                $32.97
              </span>
            </th>
          </tr>
          <tr
            class="StyledTable__StyledTableRow-sc-1m3u5g-2"
          >
            <th
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 jbJGGF"
              scope="row"
            >
              <button
                class="StyledButton-sc-323bzc-0 dYCPbF"
                type="button"
              >
                <svg
                  aria-label="FormClose"
                  class="StyledIcon-ofa7kd-0 gkyWxk"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7,7 L17,17 M7,17 L17,7"
                    fill="none"
                    stroke="#000"
                    stroke-width="2"
                  />
                </svg>
              </button>
            </th>
            <th
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 jbJGGF"
              scope="row"
            >
              <div
                class="StyledBox-sc-13pk1d4-0 ldZMaz"
              >
                <img
                  class="StyledImage-ey4zx9-0 dfToQT"
                  src="/placeholder/img/food/honey-1200xauto.jpg"
                />
              </div>
            </th>
            <th
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 jbJGGF"
              scope="row"
            >
              <span
                class="StyledText-sc-1sadyjn-0 fiywiB"
              >
                Honey
              </span>
              <p
                class="StyledParagraph-tbetod-0 ca-DKMR"
              >
                Reprehenderit in sint eiusmod esse Lorem laboris cillum nostrud ut quis ipsum non duis magna. Ea sit non id qui minim do nisi tempor ad. Proident ipsum aliqua nisi magna reprehenderit deserunt exercitation veniam laborum ad est aliquip amet. Culpa ullamco id esse culpa exercitation. Velit amet aliqua cillum tempor dolor laborum laborum. Tempor do officia ex enim anim. Labore do qui est occaecat fugiat et anim id laboris.
              </p>
            </th>
            <th
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 jbJGGF"
              scope="row"
            >
              <span
                class="StyledText-sc-1sadyjn-0 fiywiB"
              >
                $200.99
              </span>
            </th>
            <th
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 jbJGGF"
              scope="row"
            >
              <div
                class="StyledBox-sc-13pk1d4-0 degYya"
              >
                <div
                  class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 flNVco"
                >
                  <input
                    autocomplete="off"
                    class="StyledTextInput-sc-1x30a0s-0 kGmGrs"
                    type="number"
                    value="3"
                  />
                </div>
              </div>
            </th>
            <th
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 jbJGGF"
              scope="row"
            >
              <span
                class="StyledText-sc-1sadyjn-0 fiywiB"
              >
                $602.97
              </span>
            </th>
          </tr>
          <tr
            class="StyledTable__StyledTableRow-sc-1m3u5g-2"
          >
            <th
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 jbJGGF"
              scope="row"
            >
              <button
                class="StyledButton-sc-323bzc-0 dYCPbF"
                type="button"
              >
                <svg
                  aria-label="FormClose"
                  class="StyledIcon-ofa7kd-0 gkyWxk"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7,7 L17,17 M7,17 L17,7"
                    fill="none"
                    stroke="#000"
                    stroke-width="2"
                  />
                </svg>
              </button>
            </th>
            <th
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 jbJGGF"
              scope="row"
            >
              <div
                class="StyledBox-sc-13pk1d4-0 ldZMaz"
              >
                <img
                  class="StyledImage-ey4zx9-0 dfToQT"
                  src="/placeholder/img/food/onion-1200xauto.jpg"
                />
              </div>
            </th>
            <th
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 jbJGGF"
              scope="row"
            >
              <span
                class="StyledText-sc-1sadyjn-0 fiywiB"
              >
                Onions
              </span>
              <p
                class="StyledParagraph-tbetod-0 ca-DKMR"
              >
                Reprehenderit in sint eiusmod esse Lorem laboris cillum nostrud ut quis ipsum non duis magna. Ea sit non id qui minim do nisi tempor ad. Proident ipsum aliqua nisi magna reprehenderit deserunt exercitation veniam laborum ad est aliquip amet. Culpa ullamco id esse culpa exercitation. Velit amet aliqua cillum tempor dolor laborum laborum. Tempor do officia ex enim anim. Labore do qui est occaecat fugiat et anim id laboris.
              </p>
            </th>
            <th
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 jbJGGF"
              scope="row"
            >
              <span
                class="StyledText-sc-1sadyjn-0 fiywiB"
              >
                $3,000.99
              </span>
            </th>
            <th
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 jbJGGF"
              scope="row"
            >
              <div
                class="StyledBox-sc-13pk1d4-0 degYya"
              >
                <div
                  class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 flNVco"
                >
                  <input
                    autocomplete="off"
                    class="StyledTextInput-sc-1x30a0s-0 kGmGrs"
                    type="number"
                    value="1"
                  />
                </div>
              </div>
            </th>
            <th
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 jbJGGF"
              scope="row"
            >
              <span
                class="StyledText-sc-1sadyjn-0 fiywiB"
              >
                $3,000.99
              </span>
            </th>
          </tr>
          <tr
            class="StyledTable__StyledTableRow-sc-1m3u5g-2"
          >
            <th
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 jbJGGF"
              scope="row"
            >
              <button
                class="StyledButton-sc-323bzc-0 dYCPbF"
                type="button"
              >
                <svg
                  aria-label="FormClose"
                  class="StyledIcon-ofa7kd-0 gkyWxk"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7,7 L17,17 M7,17 L17,7"
                    fill="none"
                    stroke="#000"
                    stroke-width="2"
                  />
                </svg>
              </button>
            </th>
            <th
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 jbJGGF"
              scope="row"
            >
              <div
                class="StyledBox-sc-13pk1d4-0 ldZMaz"
              >
                <img
                  class="StyledImage-ey4zx9-0 dfToQT"
                  src="/placeholder/img/food/strawberry-1200xauto.jpg"
                />
              </div>
            </th>
            <th
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 jbJGGF"
              scope="row"
            >
              <span
                class="StyledText-sc-1sadyjn-0 fiywiB"
              >
                Strawberries
              </span>
              <p
                class="StyledParagraph-tbetod-0 ca-DKMR"
              >
                Reprehenderit in sint eiusmod esse Lorem laboris cillum nostrud ut quis ipsum non duis magna. Ea sit non id qui minim do nisi tempor ad. Proident ipsum aliqua nisi magna reprehenderit deserunt exercitation veniam laborum ad est aliquip amet. Culpa ullamco id esse culpa exercitation. Velit amet aliqua cillum tempor dolor laborum laborum. Tempor do officia ex enim anim. Labore do qui est occaecat fugiat et anim id laboris.
              </p>
            </th>
            <th
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 jbJGGF"
              scope="row"
            >
              <span
                class="StyledText-sc-1sadyjn-0 fiywiB"
              >
                $500.99
              </span>
            </th>
            <th
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 jbJGGF"
              scope="row"
            >
              <div
                class="StyledBox-sc-13pk1d4-0 degYya"
              >
                <div
                  class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 flNVco"
                >
                  <input
                    autocomplete="off"
                    class="StyledTextInput-sc-1x30a0s-0 kGmGrs"
                    type="number"
                    value="7"
                  />
                </div>
              </div>
            </th>
            <th
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 jbJGGF"
              scope="row"
            >
              <span
                class="StyledText-sc-1sadyjn-0 fiywiB"
              >
                $3,506.93
              </span>
            </th>
          </tr>
          <tr
            class="StyledTable__StyledTableRow-sc-1m3u5g-2"
          >
            <th
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 jbJGGF"
              scope="row"
            >
              <button
                class="StyledButton-sc-323bzc-0 dYCPbF"
                type="button"
              >
                <svg
                  aria-label="FormClose"
                  class="StyledIcon-ofa7kd-0 gkyWxk"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7,7 L17,17 M7,17 L17,7"
                    fill="none"
                    stroke="#000"
                    stroke-width="2"
                  />
                </svg>
              </button>
            </th>
            <th
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 jbJGGF"
              scope="row"
            >
              <div
                class="StyledBox-sc-13pk1d4-0 ldZMaz"
              >
                <img
                  class="StyledImage-ey4zx9-0 dfToQT"
                  src="/placeholder/img/food/tea-1200xauto.jpg"
                />
              </div>
            </th>
            <th
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 jbJGGF"
              scope="row"
            >
              <span
                class="StyledText-sc-1sadyjn-0 fiywiB"
              >
                Tea
              </span>
              <p
                class="StyledParagraph-tbetod-0 ca-DKMR"
              >
                Reprehenderit in sint eiusmod esse Lorem laboris cillum nostrud ut quis ipsum non duis magna. Ea sit non id qui minim do nisi tempor ad. Proident ipsum aliqua nisi magna reprehenderit deserunt exercitation veniam laborum ad est aliquip amet. Culpa ullamco id esse culpa exercitation. Velit amet aliqua cillum tempor dolor laborum laborum. Tempor do officia ex enim anim. Labore do qui est occaecat fugiat et anim id laboris.
              </p>
            </th>
            <th
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 jbJGGF"
              scope="row"
            >
              <span
                class="StyledText-sc-1sadyjn-0 fiywiB"
              >
                $6,000.99
              </span>
            </th>
            <th
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 jbJGGF"
              scope="row"
            >
              <div
                class="StyledBox-sc-13pk1d4-0 degYya"
              >
                <div
                  class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 flNVco"
                >
                  <input
                    autocomplete="off"
                    class="StyledTextInput-sc-1x30a0s-0 kGmGrs"
                    type="number"
                    value="2"
                  />
                </div>
              </div>
            </th>
            <th
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 jbJGGF"
              scope="row"
            >
              <span
                class="StyledText-sc-1sadyjn-0 fiywiB"
              >
                $12,001.98
              </span>
            </th>
          </tr>
        </tbody>
        <tfoot
          class="StyledTable__StyledTableFooter-sc-1m3u5g-5"
        >
          <tr
            class="StyledTable__StyledTableRow-sc-1m3u5g-2"
          >
            <td
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 cKSZRz"
            />
            <td
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 cKSZRz"
            />
            <td
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 cKSZRz"
            />
            <td
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 cKSZRz"
            />
            <td
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 kvJfCL"
            >
              <span
                class="StyledText-sc-1sadyjn-0 fiywiB"
              >
                Total
              </span>
            </td>
            <td
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 cKSZRz"
            >
              <span
                class="StyledText-sc-1sadyjn-0 fiywiB"
              >
                $19,145.84
              </span>
            </td>
          </tr>
        </tfoot>
      </table>
    </div>
  </div>
</div>
`;
